### PR TITLE
Improves detection for Samsung

### PR DESCRIPTION
**Pull request by liviuconcioiu**

### CODE REVIEW

**Comment by liviuconcioiu** [comment]
Please don't merge this PR because I want to add a few more commits. Also, I've noticed multiple rules like `SM-G988(?:0BNQUW]|U1)`. `U1` must be placed before `[0BNQUW]`. Should I correct those rules?

File: Tests/fixtures/phablet.yml [hunk 1]
```diff
@@ -5936,3 +5936,21 @@
...
+    version: "4.3"
+    platform: ""
+  client:
+    type: browser
+    name: Android Browser
+    version: ""
+    engine: WebKit
+    engine_version: ""
+  device:
+    type: phablet
```
Comment (sanchezzzhak) [inline Tests/fixtures/phablet.yml:5952]: type: tablet
size screen: 7.0
https://whatismyphone.com/vendors/samsung/galaxy-tabq-sm-t2519
https://www.devicespecifications.com/en/model/bf5d2e36

**Review by sanchezzzhak — CHANGES_REQUESTED** [review]

File: Tests/fixtures/phablet.yml [hunk 2]
```diff
@@ -5398,3 +5398,559 @@
...
+    engine: Blink
+    engine_version: ""
+  device:
+    type: phablet
+    brand: Samsung
+    model: Galaxy Z Fold 2 5G
+  os_family: Android
+  browser_family: Chrome
+-
+  user_agent: Mozilla/5.0 (Linux; Android 10; SM-F9160) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/86.0.4240.99 Safari/537.36
```
Comment (sanchezzzhak) [inline Tests/fixtures/phablet.yml:5904]: this is a duplicate of the test, the above is exactly the same test

**Comment by liviuconcioiu** [comment]
@sanchezzzhak I've seen models written as `Galaxy Tab S7+ 12.4" 5G`. Shouldn't this be written as `Galaxy Tab S7+ 5G 12.4"` instead? The same goes for `Galaxy Tab S6 10.4" Lite WiFi`, which should be `Galaxy Tab S6 Lite WiFi 10.4"`.

**Comment by sanchezzzhak** [comment]
@liviuconcioiu It's hard to say.

Most Samsung table models are named in this format 

tablet  - 
`model name` +  `(screen size)` +  `(year)`  + `network type (3G|4G|5G|WiFi|LTE)`

I think it would be better to leave it as it is
pass `Galaxy Tab S7+ 12.4" 5G`
rename `Galaxy Tab S6 10.4" Lite WiFi` in `Galaxy Tab S6 Lite 10.4" WiFi`

**Comment by liviuconcioiu** [comment]
If so, then `Galaxy Tab A 8.0" LTE (2018)` should be changed to `Galaxy Tab A 8.0" (2018) LTE` right?

File: Tests/fixtures/tablet-1.yml [hunk 3]
```diff
@@ -9293,3 +9293,845 @@
...
+  client:
+    type: browser
+    name: Samsung Browser
+    version: "11.2"
+    engine: WebKit
+    engine_version: "537.36"
+  device:
+    type: tablet
+    brand: Samsung
+    model: Galaxy Tab Active 2 8.0"
```
Comment (sanchezzzhak) [inline Tests/fixtures/tablet-1.yml:9383]: `Galaxy Tab Active 2 8.0" WiFi`

https://www.samsung.com/sg/business/tablets/galaxy-tab-active-2-8-0-t390/sm-t390nzkaxsp/

File: Tests/fixtures/smartphone-2.yml [hunk 4]
```diff
@@ -10052,3 +10052,2209 @@
...
+  client:
+    type: browser
+    name: Android Browser
+    version: ""
+    engine: WebKit
+    engine_version: ""
+  device:
+    type: smartphone
+    brand: Samsung
+    model: Galaxy Feel
```
Comment (sanchezzzhak) [inline Tests/fixtures/smartphone-2.yml:11556]: `Galaxy Feel 2`

File: Tests/fixtures/smartphone-2.yml [hunk 5]
```diff
@@ -10052,3 +10052,2209 @@
...
+  client:
+    type: browser
+    name: Samsung Browser
+    version: "11.1"
+    engine: WebKit
+    engine_version: "537.36"
+  device:
+    type: smartphone
+    brand: Samsung
+    model: Galaxy S20
```
Comment (sanchezzzhak) [inline Tests/fixtures/smartphone-2.yml:11610]: `Galaxy S20 5G`

**Review by sanchezzzhak — APPROVED** [review]

**Pull request merged**